apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/[resourceType]/[resource]/index.vue

```diff
@@ -0,0 +1,181 @@
+<script setup lang="ts">
+import { LocationQueryValue } from ".nuxt/vue-router";
+import { gql } from "graphql-request";
+import {
+  IColumn,
+  ISection,
+  ISchemaMetaData,
+  ITableMetaData,
+} from "interfaces/types";
+const config = useRuntimeConfig();
+const route = useRoute();
+const resourceName: string = route.params.resourceType as string;
+const schemaName = route.params.schema.toString();
+const metadata = await fetchMetadata(schemaName);
+
+const tableMetaDataFinderResult = metadata.tables.find(
+  (t: ITableMetaData) =>
+    t.id.toLocaleLowerCase() === resourceName.toLocaleLowerCase()
+);
+
+const tableMetaData = computed(() => {
+  if (tableMetaDataFinderResult) {
+    return tableMetaDataFinderResult;
+  } else {
+    throw new Error(`Table metadata not found for ${resourceName}`);
+  }
+});
+const resourceType = tableMetaData.value.id;
+const externalSchemaNames: string[] = extractExternalSchemas(metadata);
+const externalSchemas = await Promise.all(
+  externalSchemaNames.map(fetchMetadata)
+);
+const schemas = externalSchemas.reduce(
```

Comment:
urgently get client code as dep right?

Reply:
i this the way it is done here is better; the page ( connected to the user request) coordinates between the the async data fetching and the sync rendering of the components on the page. this imo is beter the the components client that. So yes move the ssr fetch and metadata stuff te separate module and use it